lib/text/lrc_text_parser.js

```diff
@@ -0,0 +1,121 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.text.LrcTextParser');
+
+goog.require('goog.asserts');
+goog.require('shaka.log');
+goog.require('shaka.text.Cue');
+goog.require('shaka.text.TextEngine');
+goog.require('shaka.util.StringUtils');
+
+
+/**
+ * @implements {shaka.extern.TextParser}
+ * @export
+ */
+shaka.text.LrcTextParser = class {
+  /**
+   * @override
+   * @export
+   */
+  parseInit(data) {
+    goog.asserts.assert(false, 'LRC does not have init segments');
+  }
+
+  /**
+   * @override
+   * @export
+   */
+  parseMedia(data, time) {
+    return shaka.text.LrcTextParser.getCues(data);
+  }
+
+  /**
+   * @param {BufferSource} data
+   * @return {!Array.<!shaka.extern.Cue>}
+   * @export
+   */
+  static getCues(data) {
+    const StringUtils = shaka.util.StringUtils;
+    const LrcTextParser = shaka.text.LrcTextParser;
+
+    // Get the input as a string.
+    const str = StringUtils.fromUTF8(data);
+
+    /** @type {shaka.extern.Cue} */
+    let prevCue = null;
+
+    /** @type {!Array.<!shaka.extern.Cue>} */
+    const cues = [];
+    const parts = str.split(/\r?\n/);
+    for (const part of parts) {
+      if (!part || /^\s+$/.test(part)) {
+        continue;
+      }
+
+      // LRC content
+      const match = LrcTextParser.lyricLine_.exec(part);
+      if (match) {
+        const startTime = LrcTextParser.parseTime_(match[1]);
+        // By default we add 2 seconds of duration.
+        const endTime = startTime + 2;
```

Comment:
I see that below, you adjust the end of the previous cue to match the start of this cue.  I think that's the right thing to do, but I think the final cue should last until the end of the segment instead of start + 2.  In parseMedia, the `time` parameter has a `segmentEnd` field.  I recommend you use that here.

Also, I initially missed the fact that you adjust end times below.  It would be good to mention that in a comment here.

Reply:
Done!